frontend/src/views/Announcements/AnnouncementsList.tsx

```diff
@@ -0,0 +1,213 @@
+  const { role } = useAuth();
+  const { showErrorSnackbar, showSuccessSnackbar } = useCommonSnackbars();
+  const makeRequest = useMakeRequestWithAuth();
+  const {
+    setLoading: setDeleteModalLoading,
+    isLoading: isDeleteModalLoading,
+    isOpen: isDeleteModalOpen,
+    openModal: openDeleteModal,
+    closeModal: closeDeleteModal,
+  } = useActionModal();
+
+  const [announcementToBeDeleted, setAnnouncementToBeDeleted] =
+    useState<number>();
+  const [announcementToBeEdited, setAnnouncementToBeEdited] =
+    useState<DtoAnnouncement>();
+  const [isCreateAnnouncementModalOpen, setIsCreateAnnouncementModalOpen] =
+    useState(false);
+  const [isEditAnnouncementModalOpen, setIsEditAnnouncementModalOpen] =
+    useState(false);
+
+  const handleDelete = (id: number) => {
+    openDeleteModal();
+    setAnnouncementToBeDeleted(id);
+  };
+
+  const deleteAnnouncement = async () => {
+    setDeleteModalLoading(true);
+    const announcementsApiUrl = `/api/announcements/${announcementToBeDeleted}`;
+    const response = await makeRequest<AnnouncementDeleteResponseDto>(
+      announcementsApiUrl,
+      'DELETE',
+    );
+    setDeleteModalLoading(false);
+    if (!response.ok) {
+      showErrorSnackbar();
+      return;
+    }
+    setDeleteModalLoading(false);
+    closeDeleteModal();
+    await mutate();
+    showSuccessSnackbar('Ogłoszenie zostało pomyślnie usunięte');
+  };
+
+  const handleCreate = () => {
+    setIsCreateAnnouncementModalOpen(true);
+  };
+
+  const createAnnouncement = async (values: AnnouncementFormSchema) => {
+    const { subject, body } = values;
+    const announcement: AnnouncementCreateRequestDto = {
+      announcement: {
+        subject,
+        body,
+      },
+    };
+    const announcementApiUrl = '/api/announcements';
+    const response = await makeRequest<
+      AnnouncementCreateResponseDto,
+      AnnouncementCreateRequestDto
+    >(announcementApiUrl, 'POST', announcement);
+    if (!response.ok) {
+      showErrorSnackbar();
+    }
+    showSuccessSnackbar('Ogłoszenie zostało utworzone');
+    setIsCreateAnnouncementModalOpen(false);
+    await mutate();
+  };
+
+  const handleEdit = (announcement: DtoAnnouncement) => {
+    setAnnouncementToBeEdited(announcement);
+    setIsEditAnnouncementModalOpen(true);
+  };
+
+  const editAnnouncement = async (values: AnnouncementFormSchema) => {
+    const { subject, body } = values;
+    const announcement: AnnouncementUpdateRequestDto = {
+      announcement: {
+        subject,
+        body,
+      },
+    };
+    const announcementApiUrl = `/api/announcements/${announcementToBeEdited?.id}`;
+    const response = await makeRequest<
+      AnnouncementUpdateResponseDto,
+      AnnouncementUpdateRequestDto
+    >(announcementApiUrl, 'PATCH', announcement);
+    if (!response.ok) {
+      showErrorSnackbar();
+    }
+    showSuccessSnackbar('Ogłoszenie zostało zapisane');
+    setIsEditAnnouncementModalOpen(false);
+    await mutate();
```

Comment:
Let's move mutation before the 
```suggestion
    await mutate();
    showSuccessSnackbar('Ogłoszenie zostało zapisane');
    setIsEditAnnouncementModalOpen(false);
```